students/Pirouz_N/lesson04/mailroom.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python3
+"""
+Purpose: Mailroom Part 2 python certificate from UW
+Author: Pirouz Naghavi
+Date: 07/07/2020
+"""
+
+# imports
+import datetime
+
+donor_db = {}
+
+
+def write_db_to_file():
+    """"This function saves donors to the database in donor_db.txt located at current working sub directory data."""
+    with open('data/donor_db.txt', 'w') as output_file:
+        for db_key in donor_db:
+            output_file.write(('{}: ' + '{}, ' * (len(donor_db[db_key]) - 1) + '{}\n')
+                              .format(db_key, *donor_db[db_key]))
+            # Or using f string
+            # output_file.write(f'{db_key}: {str(donor_db[db_key])[1:len(str(donor_db[db_key])) - 1]}\n')
+
+
+def populate_db_from_file():
+    """"This function reads donors from database in donor_db.txt located at current working sub directory data."""
+    try:
+        with open('data/donor_db.txt', 'r') as input_file:
+            for line in input_file:
+                colon_sep_list = line.split(':')
+                donor_db[colon_sep_list[0]] = [float(item.strip()) for item in colon_sep_list[1].strip().split(',')]
+    except TypeError:
+        print('Database data has been corrupted. Please restore a previously functioning backup.')
+    except FileNotFoundError:
+        print('Database file is no available in the current working directory folder.')
+
+
+def exit_loop():
+    """"This function allows the user to exit main menu using global condition."""
+    global continue_loop
+    continue_loop = False
```

Comment:
you can simply replace this with sys.exit() call (see mailroom guidelines) 

continue_loop global variable overcomplicates the code.

Reply:
Fixed it thank you!

Best Regards
Pirouz Naghavi